test/e2e/performanceprofile/functests/utils/node_inspector/inspector.go

```diff
@@ -19,14 +20,30 @@ import (
 	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
 	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
 	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/daemonset"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/images"
 	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/namespaces"
 	testpods "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/pods"
 )
 
 const serviceAccountSuffix = "sa"
 const clusterRoleSuffix = "cr"
 const clusterRoleBindingSuffix = "crb"
 
+func init() {
```

Comment:
Add a global boolean `initilized` if true, skip the creation

Reply:
But the `init` function is going to be called only once per process (as each suite runs as a separate process meaning the initialized value will be zeroed every suite we run).
The only benefit I see from your suggestion would be to check if `initialized` is true in the create function prior to execution, but we don't use the create function anywhere else except the `init` function.

let me know if you would still want the change, or if I miss something.

Reply:
I would still like to have a lazy initialization so the creation will happen only at the first time when we call `node.ExecCommand` and then we'll check the `initialized` value.

The advantage is for example when you're using `ginkgo.focus` for specific test that you would like to run on which you're not calling `node.ExecCommand`. If the specific test reside in a package which import the inspector, it would create the daemonset although it's not needed.

Reply:
Then what you are suggesting is to not use an init function, and just add an initialized var that will be check upon exec command.

Reply:
Yes. But I would like to get a second opinion. @ffromani WDYT?

Reply:
I personally dislike usage of `init` function for everything but the most trivial things, like setting few variables. Creating objects in the cluster as side effect of initialization  is likely to backfire in the future, composes poorly and in general is a pretty rich side effect.
I'd actually go as far as having an explicit `Setup` (or `Init` or `New` or whatever) function, but a lazy initialization like @Tal-or suggested seems a good middleground